src/networking.c

```diff
@@ -828,11 +829,15 @@ void addReplyDouble(client *c, double d) {
              sbuf[MAX_LONG_DOUBLE_CHARS+32];
         int dlen, slen;
         if (c->resp == 2) {
-            dlen = snprintf(dbuf,sizeof(dbuf),"%.17g",d);
+            dlen = fpconv_dtoa(d, dbuf);
+            dbuf[dlen] = '\0';
             slen = snprintf(sbuf,sizeof(sbuf),"$%d\r\n%s\r\n",dlen,dbuf);
             addReplyProto(c,sbuf,slen);
         } else {
-            dlen = snprintf(dbuf,sizeof(dbuf),",%.17g\r\n",d);
+            dlen = fpconv_dtoa(d, dbuf);
+            dbuf[dlen] = '\r';
+            dbuf[dlen+1] = '\n';
+            dbuf[dlen+2] = '\0';
```

Comment:
You forget the top `,` and update dlen.
```suggestion
            dbuf[0] = ',';
            dlen = fpconv_dtoa(d, dbuf+1);
            dbuf[dlen] = '\r';
            dbuf[dlen+1] = '\n';
            dbuf[dlen+2] = '\0';
            dbuf += 3;
```

Reply:
My bad, `dbuf += 3;` should be `dlen += 3`.